src/Data/MIME.hs

```diff
@@ -734,7 +734,8 @@ textCharsetSources =
 
   -- https://tools.ietf.org/html/rfc2854
   -- The default is ambiguous; using us-ascii for now
-  , ("html", InPayloadOrParameter (\_param _payload -> Just "us-ascii")) -- FIXME
+  -- TODO: perhaps analyze html content to determine charset
+  , ("html", InPayloadOrParameter (\param _payload -> if isJust param then param else Just "us-ascii"))
```

Comment:
```
 , ("html", InPayloadOrParameter (\param _payload -> param <|> Just "us-ascii"))
```
Minor improvement that uses the Alternative instance for Maybe, but your version works too. You can ignore this one.